File: 프로그래머스/lv2/42747. H－Index/H－Index.java
import java.util.Arrays;
class Solution {
    public int solution(int[] citations) {
        Arrays.sort(citations);
         int result = 1;
        while(true){
      int temp = 0;
      for(int i=0; i<citations.length; i++){
        if(citations[i] >= result){
          temp++;
        }
      } 
      if(result == temp) break;
      if( result > temp) {
        result--; 
        break;
      }
      else result++;
      
    }
    
        
        return result;
    }
    
}

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    let answer = 0;
    
    citations.sort((a, b) => a - b);
    
    const N = citations.length;
    const mid = Math.floor(N / 2);
    
    for (let i = 0; i < N; i++) {
        if (citations[i] >= N - i) {
            answer = N - i;
            break;
        }
    }
    
    
    return answer;
}